janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/NumericTranslationQueryParser.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.lucene.document.LongPoint;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+import org.janusgraph.diskstorage.indexing.KeyInformation;
+import org.janusgraph.graphdb.database.serialize.AttributeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NumericTranslationQueryParser extends QueryParser {
+    private static final Logger log = LoggerFactory.getLogger(NumericTranslationQueryParser.class);
+    private final KeyInformation.StoreRetriever storeRetriever;
+
+    public NumericTranslationQueryParser(KeyInformation.StoreRetriever storeRetriever, String field, Analyzer analyzer) {
+        super(field, analyzer);
+        this.storeRetriever = storeRetriever;
+    }
+
+    @Override
+    protected Query newRangeQuery(final String field, final String start, final String end, final boolean startInclusive,
+                                  final boolean endInclusive) {
+
+        Class<?> dataType = storeRetriever.get(field).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return getNumericRangeQuery(field, dataType, start, end, startInclusive, endInclusive);
+            } catch (NumberFormatException e) {
+                printNumberFormatException(field, dataType, e);
+            }
+        }
+
+        return super.newRangeQuery(field, start, end, startInclusive, endInclusive);
+    }
+
+    @Override
+    protected Query getRangeQuery(final String field, final String start, final String end, final boolean includeLower,
+                                  final boolean includeUpper) throws ParseException {
+        Class<?> dataType = storeRetriever.get(field).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return getNumericRangeQuery(field, dataType, start, end, includeLower, includeUpper);
+            } catch (NumberFormatException e) {
+                printNumberFormatException(field, dataType, e);
+            }
+        }
+
+        return super.getRangeQuery(field, start, end, includeLower, includeUpper);
+    }
+
+    @Override
+    protected Query newTermQuery(Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
+        }
+
+        Class<?> dataType = storeRetriever.get(t.field()).getDataType();
+        if (Number.class.isAssignableFrom(dataType)) {
+            try {
+                return buildNumericQuery(t.field(), t.text(), dataType);
+            } catch (NumberFormatException e) {
+                printNumberFormatException(t.field(), dataType, e);
+            }
+        }
+
+        return super.newTermQuery(t);
+    }
+
+    @Override
+    protected Query newWildcardQuery(final Term t) {
+        if (t.field() == null) {
+            return super.newTermQuery(t);
```

Comment:
Should this be `super.newWildcardQuery(t)`?

Reply:
FIxed